core/src/main/java/io/grpc/internal/CompositeReadableBuffer.java

```diff
@@ -50,9 +58,9 @@ public void addBuffer(ReadableBuffer buffer) {
     }
 
     CompositeReadableBuffer compositeBuffer = (CompositeReadableBuffer) buffer;
-    while (!compositeBuffer.buffers.isEmpty()) {
-      ReadableBuffer subBuffer = compositeBuffer.buffers.remove();
-      buffers.add(subBuffer);
+    Queue<ReadableBuffer> otherBuffers = compositeBuffer.buffers;
```

Comment:
I agree it's unlikely to make much/any difference especially here, but seems cleaner and good general practice. I.e. in theory less pointer chasing in the loop. Or could be my OCD.

Reply:
I don't think I agree with "good general practice." I expect the optimizer to do loop invariant hoisting. For simple cases like this I'd put it in the same category as 'optimizing' `/2` with `>>1` or simplifying `1 + 2` to `3`. But this isn't even that hot of a loop. With reasonably-sized buffers, it seems impossible to ever notice this optimization, even if we assume the processor isn't faster executing the later occurrences or the interpreter was being used.

The "cleaner" part for me was combining `buffers.add()` and the `remove()` onto the same line. But the original author could have done that and actively chose not to. But adding `otherBuffers` actually the change overall less clean to my eyes. In any case the original code seems reasonable as-is, so it seems best not to touch it.

Reply:
Sure, I'll revert it! Yes the "cleanliness" is probably subjective, I have a general preference for minimizing indirections, other things being equal (regardless of perf implications). I don't quite agree that it's in the same category as `>>1`, `1 + 2` -> `3` since those have a more obvious readability tradeoff.